File: 351001/Boukhvalova/module-publisher/src/main/java/by/boukhvalova/distcomp/dto/UserRequestTo.java
package by.boukhvalova.distcomp.dto;

import by.boukhvalova.distcomp.entities.UserRole;
import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class UserRequestTo {
    long id;

    @Size(min = 2, max = 64)
    String login;

    @Size(min = 8, max = 128)
    String password;

    @Size(min = 2, max = 64)
    @JsonProperty("firstname")
    @JsonAlias({"firstName"})
    String firstname;

    @Size(min = 2, max = 64)
    @JsonProperty("lastname")
    @JsonAlias({"lastName"})
    String lastname;

    UserRole role;
}

File: 351002/Mikhalkov/publisher/Distcomp.Publisher.Application/DTOs/UserRequestTo.cs
using Distcomp.Shared.Models;
using System.Text.Json.Serialization;

namespace Distcomp.Application.DTOs
{
    public record UserRequestTo(
        long? Id,
        [property: JsonPropertyName("login")] string Login,
        [property: JsonPropertyName("password")] string Password,
        [property: JsonPropertyName("firstname")] string FirstName,
        [property: JsonPropertyName("lastname")] string LastName,
        [property: JsonPropertyName("role")] UserRole Role
    );
}